inference-engine/src/readers/ir_reader/ie_ir_parser.cpp

```diff
@@ -683,9 +683,9 @@ V10Parser::GenericLayerParams V10Parser::XmlDeserializer::parseGenericParams(con
             size_t dim = 0;
             const pugi::char_t* dimVal = node.child_value();
             std::stringstream ss(dimVal);
-            if (!(ss >> dim) || dim == 0) {
+            if (!(ss >> dim) || dim < 0) {
```

Comment:
Dim has size_t type which can not be negative so it will always return true. Is it expected? :)

Reply:
It is really strange but this check really works even with `size_t`! :D However, this is an obvious bug.

Anyway, I found that the model cannot be inferred even with the proper fix. I will prepare some synthetic test and update the PR.